src/service.ts

```diff
@@ -261,36 +266,13 @@ export class Service {
         }, this.embeds);
 
         if(embed) {
-            utils.raiseCustomEvent(embed.element, event.name, event.value);
-        }
-    }
+            let value = event.value;
 
-    /**
-     * Translate target into url
-     * Target may be to the whole report, speific page, or specific visual
-     */
-    private getTargetUrl(target?: models.IPageTarget | models.IVisualTarget): string {
-        let targetUrl;
+            if (event.name === 'pageChanged') {
+                value.page = new Page(embed, event.value.page.name);
```

Comment:
This is the weird part about require event data to be instances of Pages classes instead normal page objects.

It feels awkward to have logic specific to different types of events instead of treating them uniformly.